File: VB/DxSample/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports DevExpress.Diagram.Core

Namespace DxSample

    Public Partial Class Form1
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Protected Overrides Sub OnLoad(ByVal e As EventArgs)
            MyBase.OnLoad(e)
            Call RegisterCustomShapes()
            diagramControl1.InitializeRibbon(ribbonControl1)
        End Sub

        Private Shared Sub RegisterCustomShapes()
            Dim dict = New System.Windows.ResourceDictionary()
            dict.Source = New Uri("DiagramResources.xaml", UriKind.RelativeOrAbsolute)
            DiagramToolboxRegistrator.RegisterShapes("CustomShapes", Function() "Custom Shapes", dict, Function(id) id)
        End Sub
    End Class
End Namespace

File: CS/DxSample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.Diagram.Core;

namespace DxSample {
    public partial class Form1 : DevExpress.XtraBars.Ribbon.RibbonForm {
        public Form1() {
            InitializeComponent();
        }
        protected override void OnLoad(EventArgs e) {
            base.OnLoad(e);
            RegisterCustomShapes();
            this.diagramControl1.InitializeRibbon(this.ribbonControl1);
        }

        static void RegisterCustomShapes() {
            var dict = new System.Windows.ResourceDictionary();
            dict.Source = new Uri("DiagramResources.xaml", UriKind.RelativeOrAbsolute);
            DiagramToolboxRegistrator.RegisterShapes("CustomShapes", () => "Custom Shapes", dict, id => id);
        }
    }
}
